src/main/java/org/datadog/jenkins/plugins/datadog/clients/DatadogAgentClient.java

```diff
@@ -441,6 +420,21 @@ public void gauge(String name, long value, String hostname, Map<String, Set<Stri
             }
         }
 
+        @Override
+        public void rate(String name, double value, String hostname, Map<String, Set<String>> tags) {
+            try {
+                boolean status = reinitializeStatsDClient(false);
+                if(!status){
+                    return;
+                }
+                logger.fine("increment counter with dogStatD client");
+                statsd.count(name, value, TagsUtil.convertTagsToArray(tags));
```

Comment:
```suggestion
                String[] tagsArr = (tags != null) ? TagsUtil.convertTagsToArray(tags) : EMPTY_ARRAY;
                statsd.count(name, value, TagsUtil.convertTagsToArray(tagsArr));
```
Or even better, modify the `TagsUtil.convertTagsToArray` method to be resilient for `null` tags.
In the `DatadogApiClient` class, we're checking for this.

Reply:
Added a null-check in `TagsUtil.convertTagsToArray`